packages/machinomy/src/PaymentRequiredRequest.ts

```diff
@@ -0,0 +1,33 @@
+import { InvalidUrl } from './Exceptions'
+
+export class PaymentRequiredRequest {
+  datetime?: number
+  constructor (datetime?: number) {
+    this.datetime = datetime
+  }
+}
+
+export class PaymentRequiredRequestSerializer {
+  static instance: PaymentRequiredRequestSerializer = new PaymentRequiredRequestSerializer()
+
+  serialize (obj: PaymentRequiredRequest, baseurl: string): string {
+    const url = `${baseurl}${obj.datetime ? `/${obj.datetime}` : ''}`
```

Comment:
Why date is required here?

Reply:
this parameter is optional. We ask the price for payment at a specific moment, if the date is not determined - I think that this is an "abstract price", because the price is valid for a certain period of time. Used for shopping in the past.